source/nvda.pyw

```diff
@@ -213,12 +236,82 @@ def getDesktopName():
+desktopName = getDesktopName()
+_log.info(f"DesktopName: {desktopName}")
+
+
+def _acquireMutex(_desktopName: str) -> typing.Optional[wintypes.HANDLE]:
+	# From MS docs; "Multiple processes can have handles of the same mutex object"
+	# > Two or more processes can call CreateMutex to create the same named mutex.
+	# > The first process actually creates the mutex, and subsequent processes with sufficient access rights
+	# > simply open a handle to the existing mutex.
+	# > This enables multiple processes to get handles of the same mutex, while relieving the user of the
+	# > responsibility of ensuring that the creating process is started first.
+	# > When using this technique, you should set the bInitialOwner flag to FALSE; otherwise, it can be difficult
+	# > to be certain which process has initial ownership.
+	# > https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-createmutexw
+	_mutex = ctypes.windll.kernel32.CreateMutexW(
+		None,  # lpMutexAttributes,
+		# Don't take initial ownership, use wait to acquire ownership instead.
+		# Allows waiting for a prior process to finish exiting.
+		False,  # bInitialOwner
+		f"Local\\NVDA_{_desktopName}"  # lpName
+	)
+	createMutexResult = ctypes.windll.kernel32.GetLastError()
+	if not _mutex:
+		_log.error(f"Unable to create mutex, last error: {createMutexResult}")
+		raise winUser.WinError(createMutexResult)
+	else:
+		if createMutexResult == winKernel.ERROR_ALREADY_EXISTS:
+			_log.debug("Waiting for prior NVDA to finish exiting")
+		# We didn't ask to be the initial owner,
+		waitResult = winKernel.waitForSingleObject(
+			_mutex,  # hHandle
+			2000  # dwMilliseconds
+		)
+
+		_log.debug(f"Wait result: {waitResult}")
+		if winKernel.WAIT_OBJECT_0 == waitResult:
+			_log.info("Prior NVDA has finished exiting")
+			return _mutex  # mutex ownership acquired
+		elif winKernel.WAIT_ABANDONED == waitResult:
+			_log.error(
+				"Prior NVDA exited without releasing mutex, taking ownership."
+				" Note: Restarting your system is recommended."
+				" This error indicates that NVDA previously did not exit correctly or was terminated"
+				" (perhaps by the task manager)."
+			)
+			return _mutex  # mutex ownership acquired
+		else:
+			exception = None
+			if winKernel.WAIT_TIMEOUT == waitResult:
+				exception = Exception("Timeout exceeded waiting for mutex")
+			elif winKernel.WAIT_FAILED == waitResult:
+				waitError = winUser.GetLastError()
+				_log.debug(f"Failed waiting for mutex, error: {waitError}")
+				exception = winUser.WinError(waitError)
+			releaseResult = ctypes.windll.kernel32.ReleaseMutex(_mutex)
+			if 0 == releaseResult:
+				releaseError = winUser.GetLastError()
+				_log.debug(f"Failed to release mutex, error: {releaseError}")
+			closeResult = ctypes.windll.kernel32.CloseHandle(_mutex)
+			if 0 == closeResult:
+				closeError = winUser.GetLastError()
+				_log.debug(f"Failed to close mutex handle, error: {closeError}")

```

Comment:
nitpicky - but should this repeated code to release and close mutex be abstracted into a function? I've replied to the duplicate section as well.

Reply:
Potentially, note they are currently using different loggers. Given the time it takes to test this, I think I'll leave it as is.